packages/sls-env/src/index.ts

```diff
@@ -31,7 +33,13 @@ import type {
  * @template R - a result that application will return
  *
  */
-type SlsEnvironment<H extends Handler<any, any, any>, C, D, P = HandlerPayload<H>, R = ReturnType<H>> = {
+type SlsEnvironment<
+  H extends Handler<any, any, any>,
+  C,
+  D extends Record<string, any>,
```

Comment:
`Record` does not look like a valid type for `dependencies`. I could be wrong but don't think that work nicely with `application` especially when you pass `eventBasedDependency`. 

You don't want application to know about `eventBasedDependency` as that has to be opaque. IMO we need to do types infring to get the actual correct type.